test/common/buffer/zero_copy_input_stream_test.cc

```diff
@@ -90,6 +90,147 @@ TEST_F(ZeroCopyInputStreamTest, Finish) {
   EXPECT_FALSE(stream_.Next(&data_, &size_));
 }
 
+class ZeroCopyInputStreamSkipTest : public testing::Test {
+public:
+  ZeroCopyInputStreamSkipTest() {
+    Buffer::OwnedImpl buffer;
+    buffer.addBufferFragment(buffer1_);
+    buffer.addBufferFragment(buffer2_);
+    buffer.addBufferFragment(buffer3_);
+    buffer.addBufferFragment(buffer4_);
+
+    stream_.move(buffer);
+  }
+
+  const std::string slice1_{"This is the first slice of the message."};
+  const std::string slice2_{"This is the second slice of the message."};
+  const std::string slice3_{"This is the third slice of the message."};
+  const std::string slice4_{"This is the fourth slice of the message."};
+  BufferFragmentImpl buffer1_{slice1_.data(), slice1_.size(), nullptr};
+  BufferFragmentImpl buffer2_{slice2_.data(), slice2_.size(), nullptr};
+  BufferFragmentImpl buffer3_{slice3_.data(), slice3_.size(), nullptr};
+  BufferFragmentImpl buffer4_{slice4_.data(), slice4_.size(), nullptr};
+
+  const size_t total_bytes_{slice1_.size() + slice2_.size() + slice3_.size() + slice4_.size()};
+  ZeroCopyInputStreamImpl stream_;
+
+  const void* data_;
+  int size_;
+};
+
+TEST_F(ZeroCopyInputStreamSkipTest, SkipFirstPartialSlice) {
+  // Only skip the 10 bytes in the first slice.
+  const int SkipCount = 10;
+  EXPECT_TRUE(stream_.Skip(SkipCount));
+
+  EXPECT_EQ(SkipCount, stream_.ByteCount());
+
+  // Read the first slice
+  EXPECT_TRUE(stream_.Next(&data_, &size_));
+  EXPECT_EQ(slice1_.size() - SkipCount, size_);
+  EXPECT_EQ(slice1_.substr(SkipCount), std::string(reinterpret_cast<const char*>(data_), size_));
```

Comment:
have `data_` as `const char*` then you don't need casts here and below?

Reply:
I couldn't.  stream_.Next() is expecting `const void**`.  
I added a data_string() function to convert it in one place.